simulation/vis_simulate.pro

```diff
@@ -238,15 +250,17 @@ FUNCTION vis_simulate,obs,status_str,psf,params,jones,skymodel,file_path_fhd=fil
           
           if max(abs(*this_model_uv[0])) eq 0 and max(abs(*this_model_uv[1])) eq 0 then continue
           
-          this_model_ptr=vis_source_model(skymodel,obs,status_str,psf,params,this_vis_weight_ptr,model_uv_arr=this_model_uv,$
-            timing=model_timing,silent=silent,error=error,_Extra=extra)
-        ;FOR pol_i=0,n_pol-1 DO BEGIN
-         ;   vis_model_arr[pol_i]=visibility_degrid(*this_model_uv[pol_i],this_vis_weight_ptr[pol_i],obs,psf,params,silent=silent,$
-         ;       polarization=pol_i,_Extra=extra)
-        ;ENDFOR
+        ;  this_model_ptr=vis_source_model(skymodel,obs,status_str,psf,params,this_vis_weight_ptr,model_uv_arr=this_model_uv,$
+        ;    timing=model_timing,silent=silent,error=error,_Extra=extra)
+          model_timing=0.0
+          FOR pol_i=0,n_pol-1 DO BEGIN
+              (*vis_model_arr[pol_i])[fi,*] = (*(visibility_degrid(*this_model_uv[pol_i],this_vis_weight_ptr[pol_i],obs,psf,params,timing=timing,silent=silent,$
+                  polarization=pol_i,_Extra=extra)))[fi,*]
+               model_timing += timing
+          ENDFOR    ; vis_source_model uses DFT, which works for point source models. Since the uvf cube was already FFTed, just run visibility_degrid.
```

Comment:
Please move this comment up to the start of the new block of code. I don't like leaving commented-out code in, but that was there before and is not your responsibility to fix right now.

Reply:
Move the comment "vis_source_model uses DFT..." up to line 256?